src/searchModal.js

```diff
@@ -419,13 +441,51 @@ export default function searchModalFactory(config) {
      */
     function buildDataModel(data) {
         if (data.settings) {
+            //save availableColumns to memory
+            availableColumns = data.settings.availableColumns;
             // @todo: use the selected columns instead. It can use a promise as it takes place insise a promise chain
-            data.model = columnsToModel(data.settings.availableColumns);
+            data.model = columnsToModel(availableColumns);
+            dataCache = _.cloneDeep(data);
+            return data;
         } else {
             data.model = columnsToModel(_.values(data.model));
+            dataCache = _.cloneDeep(data);
+            return data;
         }
+    }
 
-        return data;
+    /**
+     * Filters datatble model based on stored selected columns
+     * @param {Object} data data containing available columns and model for datatable
+     * @returns {Promise} promise which resolves with filtered data.model
+     */
+    function filterSelectedColumns(data) {
+        console.log(data);
+        return selectedColumnsStore
+            .getItem(getClassFilterUri())
+            .then(storedSelectedColumnIds => {
+                //let columnsToDisplay;
+                if (storedSelectedColumnIds && storedSelectedColumnIds.length) {
+                    selectedColumns = [...storedSelectedColumnIds];
+                    //columnsToDisplay = data.settings.availableColumns.filter(column =>
+                    // selectedColumns.includes(column.id)
+                    //);
+                } else {
+                    selectedColumns = data.settings.availableColumns.reduce((acc, column) => {
+                        if (column.default) {
+                            acc.push(column.id);
+                        }
+                    }, []);
+                }
+
+                data.model = data.model.filter(column => selectedColumns.includes(column.id));
+
+                return data;
+            })
+            .catch(e => {
+                instance.trigger('error', e);
+                reject(new Error('Error getting selected columns'));
```

Comment:
**thought:** Is there a need to trigger this error? It does not seem to be caught. It would produce unresolved promise errors, but actually, there is no wrapping promise. Is it a dead code?

Reply:
actually `reject` is not defined here